### PR TITLE
Refactor build script and default to preview

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
## PR Summary

Fixes https://github.com/PowerShell/vscode-powershell/issues/1880

* Makes the build script simpler and more organized
* Updates the README for preview only during CI and release
* Updates the version to a date-based version only during CI and release
* makes "Preview" the default in the repo

## PR Checklist

Note: Tick the boxes below that apply to this pull request by putting an `x` between the square brackets.
Please mark anything not applicable to this PR `NA`.

- [x] PR has a meaningful title
- [x] Summarized changes
- [ ] PR has tests
- [x] This PR is ready to merge and is not work in progress
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready

### CODE REVIEW

File: vscode-powershell.build.ps1 [hunk 1]
```diff
@@ -119,20 +134,30 @@ task UpdateReadme {
...
+        $script:PackageJson.description = "(Preview) Develop PowerShell scripts in Visual Studio Code!"
+        $script:PackageJson.preview = $true
+    } else {
+        $script:PackageJson.name = "powershell"
+        $script:PackageJson.displayName = "PowerShell"
+        $script:PackageJson.description = "Develop PowerShell scripts in Visual Studio Code!"
+        $script:PackageJson.preview = $false
+    }
+
+    $revision = if ($env:VSTS_BUILD_VERSION) { $env:VSTS_BUILD_VERSION } else { 9999 }
```
Comment (TylerLeonhardt) [inline vscode-powershell.build.ps1]: Thoughts on using `$env:BUILD_BUILDID` instead of this `$env:VSTS_BUILD_VERSION`?

@rjmholt

Comment (TylerLeonhardt) [reply vscode-powershell.build.ps1]: There's also `System.PullRequest.PullRequestId` but I don't think that'll be defined in CI builds, only PR builds

Comment (TylerLeonhardt) [reply vscode-powershell.build.ps1]:
```suggestion
    $revision = if ($env:BUILD_BUILDID) { $env:BUILD_BUILDID } else { 9999 }
```

**Comment by TylerLeonhardt** [comment]
Codacy failures are just about `Write-Host` which we can ignore.

File: vscode-powershell.build.ps1 [hunk 2]
```diff
@@ -119,20 +134,30 @@ task UpdateReadme {
     $readmeContent = Get-Content -Path $readmePath
     if (!($readmeContent -match "This is the PREVIEW version of the PowerShell extension")) {
         $readmeContent[0] = $newReadmeTop
-        $readmeContent > $readmePath
+        $readmeContent | Set-Content $readmePath -Encoding utf8
```
[the remark this answers is not in the document]
Comment (TylerLeonhardt) [reply vscode-powershell.build.ps1]: hmm not sure what the default is for this... but I could put `utf8NoBOM`. Thoughts?

[the remark this answers is not in the document]
Comment (bergmeister) [reply vscode-powershell.build.ps1]: The docs say that `UTF8NoBOM`is already the default
https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/set-content

[the remark this answers is not in the document]
Comment (TylerLeonhardt) [reply vscode-powershell.build.ps1]: Thanks @bergmeister!

[the remark this answers is not in the document]
Comment (rjmholt) [reply vscode-powershell.build.ps1]: The default is UTF8NoBOM in PSCore, but to make it work in Windows PowerShell too I think we have to [fall back to .NET](https://stackoverflow.com/questions/5596982/using-powershell-to-write-a-file-in-utf-8-without-the-bom)

[the remark this answers is not in the document]
Comment (rjmholt) [reply vscode-powershell.build.ps1]: But we might not need this to work in Windows PowerShell

[the remark this answers is not in the document]
Comment (TylerLeonhardt) [reply vscode-powershell.build.ps1]: updated to WriteAllLines .NET API to support Windows PowerShell

[the remark this answers is not in the document]
Comment (bergmeister) [reply vscode-powershell.build.ps1]: Why can we not drop WindowsPowerShell for building purposes only? Developers are not users and I added a `pwsh` switch to the Azure DevOps task last year and it seems we are not using AppVeyor any more.
I was thinking of changing [this](https://github.com/PowerShell/vscode-powershell/blob/master/.vscode/tasks.json#L7) line here to use use `pwsh` instead.

[the remark this answers is not in the document]
Comment (rjmholt) [reply vscode-powershell.build.ps1]: I think we probably want to move to `pwsh` only for building, but I suspect currently we just pick a PowerShell and stick with it. So what @TylerLeonhardt's got is safest for now.

**Pull request merged**